op-supervisor/supervisor/backend/depset/mock.go

```diff
@@ -0,0 +1,68 @@
+package depset
+
+import (
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+type MockDependencySet struct {
```

Comment:
Why mock a dependency set when a regular static dependency set is so simple to configure? When do we need these callbacks?